addons/web/controllers/binary.py

```diff
@@ -166,11 +166,13 @@ def content_image(self, xmlid=None, model='ir.attachment', id=None, field='raw',
                       download=False, width=0, height=0, crop=False, access_token=None,
                       nocache=False):
         try:
-            record = request.env['ir.binary']._find_record(xmlid, model, id and int(id), access_token)
+            record = request.env['ir.binary']._find_record(xmlid, model, id and int(id), access_token, field=field)
             stream = request.env['ir.binary']._get_image_stream_from(
                 record, field, filename=filename, filename_field=filename_field,
                 mimetype=mimetype, width=int(width), height=int(height), crop=crop,
             )
+        except AccessError:
+            raise request.not_found()
```

Comment:
I think you want to apply the same principle as the other exc, to avoid broken images and have placeholder instead.